adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+
+The core architectural question is: How can we build a robust, developer-friendly agentic framework on a stateless API without sacrificing the usability that an explicit `Agent` construct provides?
+
+## Decision
+
+We will adopt the **Llama Stack Responses API** as the exclusive backend interface for agentic reasoning within Red Hat's AI products.
+
+To address the absence of a server-side agent object, we will **design and implement a client-side Agent construct**. This client-side agent construct will be responsible for:
+
+* **State Management:** Maintaining the conversation history (user messages, AI responses, tool outputs).
+* **Agentic Loop:** Orchestrating the turn-by-turn interaction with the Llama Stack Responses API. This involves:
+    1. Formatting the current state (chat history, available tools) into a request.
+    2. Sending the request to the API endpoint.
+    3. Parsing the API response to distinguish between textual replies and tool call requests.
+    4. Executing any requested tool calls using client-side or remote tooling.
+    5. Formatting the tool execution results.
+    6. Appending the tool results to the history and repeating the loop until a final answer is generated or a stop condition is met.
+* **Tool Dispatching:** Managing the registration and invocation of available tools on the client side.
+
+The decision of what form this client-side Agent construct will take will be addressed in a subsequent ADR. Some proposals include adding the functionality to the official Llama Stack client library, creating a new SDK maintained by Red Hat (perhaps as part of OpenDataHub), providing a container image with a simple HTTP API, and adopting a pre-existing SDK such as LangChain. That ADR will cover the pros and cons of these options and propose a final resolution.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Strategic Alignment & Future-Proofing:** By building on the OpenAI-compatible Responses API, we align with the industry standard and the strategic direction of Llama Stack. This de-risks our platform from the likely deprecation of the older Agents API.
+* **Ecosystem Compatibility:** Our agentic capability will be natively compatible with the vast ecosystem of tools, libraries, and monitoring solutions that are built to interface with the OpenAI API format. This accelerates both internal development and customer adoption.
+* **Enhanced Flexibility and Control:** A client-side agent construct gives the application developer full control over the agent's execution loop, state, and tool implementation. This is critical for complex, mission-critical applications where logic cannot be a black box on a server.
```

Comment:
True, but is it up to us to create that client-side agent framework or should we settle on an existing one?